src/by/teachmeskills/collections/Container.java

```diff
@@ -0,0 +1,82 @@
+package by.teachmeskills.collections;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class Container {
+
+    private Set<Ball> balls;
+
+    public void removeBall(Set<Ball> balls, Ball ball) {
+        this.balls = balls;
+        if (balls.contains(ball)) {
+            balls.remove(ball);
+            System.out.println(ball + " was removed from set");
+        } else {
+            System.out.println("Set does not contain " + ball);
+        }
+    }
+
+    public void addBall(Set<Ball> balls, Ball ball) {
+        this.balls = balls;
+        if (balls.contains(ball)) {
+            System.out.println(ball + " is already defined in set. Please add unique ball");
+        } else {
+            balls.add(ball);
+        }
+    }
+
+    public void addBalls(Set<Ball> balls) {
+        this.balls = balls;
+        new HashSet<>(balls);
+    }
+
+    public void getBallsAmount(Set<Ball> balls) {
+        System.out.println("Amount of balls = " + balls.size());
+    }
+
+    public void cleanUpContainer(Set<Ball> balls) {
+        balls.clear();
+    }
+
+    public void getBallIfExist(Set<Ball> balls, Ball ball) {
+        boolean matches = Stream.of(balls).anyMatch(b -> balls.contains(ball));
```

Comment:
` boolean matches = this.balls.stream()
                                    .anyMatch(b -> balls.contains(ball));`